examples/iota_lens/analysis_iotalens.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python3
+#
+# Copyright 2022 ImpactX contributors
+# Authors: Axel Huebl, Chad Mitchell
+# License: BSD-3-Clause-LBNL
+#
+
+import glob
+
+import numpy as np
+import pandas as pd
+from scipy.stats import moment
+
+
+def get_moments(beam):
+    """Calculate mean and std dev of functions defining the IOTA invariants
+
+    Returns
+    -------
+    meanH, sigH, meanI, sigI
+    """
+    meanH = np.mean(beam["H"])
+    sigH = moment(beam["H"],moment=2)**0.5
+    meanI = np.mean(beam["I"])
+    sigI = moment(beam["I"],moment=2)**0.5
+
+    return (
+        meanH, sigH, meanI, sigI)
+
+
+def read_all_files(file_pattern):
+    """Read in all CSV files from each MPI rank (and potentially OpenMP
+    thread). Concatenate into one Pandas dataframe.
+
+    Returns
+    -------
+    pandas.DataFrame
+    """
+    return pd.concat(
+        (
+            pd.read_csv(filename, delimiter=r"\s+")
+            for filename in glob.glob(file_pattern)
+        ),
+        axis=0,
+        ignore_index=True,
+    )
+
+
+# initial/final beam on rank zero
+initial = read_all_files("diags/initial_nonlinear_lens_invariants.txt.*")
+final = read_all_files("diags/output_nonlinear_lens_invariants.txt.*")
```

Comment:
@ax3l add join on particle ID with new columns:
- `dH = H_initial - H_final`
- `dI = I_initial - H_final`

Reply:
added now. tolerances to be discussed with @cemitch99 :)